recipes/botan/all/conanfile.py

```diff
@@ -117,84 +124,155 @@ def config_options(self):
 
         # --single-amalgamation option is no longer available
         # See also https://github.com/randombit/botan/pull/2246
-        if tools.Version(self.version) >= '2.14.0':
+        if Version(self.version) >= '2.14.0':
             del self.options.single_amalgamation
 
+        # Support for the OpenSSL provider was removed in 2.19.2
+        if Version(self.version) >= '2.19.2':
+            del self.options.with_openssl
+
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
         if self.options.with_bzip2:
             self.requires("bzip2/1.0.8")
-        if self.options.with_openssl:
-            self.requires("openssl/1.1.1o")
+        if self.options.get_safe('with_openssl', False):
+            self.requires("openssl/1.1.1s")
```

Comment:
```suggestion
            self.requires("openssl/[>=1.1 <4]")
```
OpenSSL and zlib can use range versions on CCI: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

Hope it supports OpenSSL 3.x

Reply:
No OpenSSL 3 is not supported by Botan

Reply:
@lieser thank you for your quick feedback. I just updated the pattern to not accept openssl 3. So we can move automatically in case having a newer version available.

Reply:
Botan 2.x can (optionally) consume OpenSSL 1.x as a crypto provider. Support for the OpenSSL provider was removed in Botan 3.x. The `with_openssl` option is not available when requesting the Botan 3.x recipe.

Hence: This is intentional.

Reply:
Nevertheless, I applied your suggestion. Auto-updating won't hurt.